### ISSUE TITLE
调用p-tuning报错

### ISSUE THREAD

**Issue by linguoqi** [modelscope/swift#340]
`import os
os.environ['CUDA_VISIBLE_DEVICES'] = '3'
from modelscope import Model, AutoModelForSequenceClassification, AutoTokenizer, MsDataset
from swift import Swift, LoRAConfig, AdapterConfig, Trainer, TrainingArguments, PromptConfig
import torch
from transformers import default_data_collator

model = Model.from_pretrained('/tf/model/Llama-2-7b-chat-ms/', torch_dtype=torch.bfloat16)
adapter_config = AdapterConfig(
                dim=model.config.hidden_size,
                target_modules=['mlp'],
                method_name='forward',
                hidden_pos=0,
                adapter_length=32,
            )

prompt_config = PromptConfig(
    dim=768,  # hidden states的维度
    target_modules=r'.*layer\.\d+$',  # 要使用正则表达式替换的模块
    embedding_pos=0,    # embedding张量的位置
    prompt_length=10,   # 提示符token的长度
    attach_front=False  # 是否将提示符附加在embedding前面
)

print(adapter_config)
model = Swift.prepare_model(model, prompt_config)
tokenizer = AutoTokenizer.from_pretrained('/tf/model/Llama-2-7b-chat-ms/')

train_dataset = MsDataset.load(dataset_name = "/tf/user/lgq/data",data_files={"train": '/tf/user/lgq/data/black_start_case14_versionB.jsonl'}).to_hf_dataset().select(range(4))
val_dataset = MsDataset.load(dataset_name = "/tf/user/lgq/data",data_files={"val": '/tf/user/lgq/data/black_start_case14_versionB.jsonl'}).to_hf_dataset().select(range(4))

print(train_dataset)
def tokenize_function(examples):
    return tokenizer(examples["instruction"], examples["input"],examples["output"], padding="max_length", truncation=True, max_length=128)

train_dataset = train_dataset.map(tokenize_function, batched=True)
val_dataset = val_dataset.map(tokenize_function, batched=True)

arguments = TrainingArguments(
    output_dir='./outputs',
    per_device_train_batch_size=4,
)

trainer = Trainer(model, arguments, train_dataset=train_dataset,
                    eval_dataset=val_dataset,
                    data_collator=default_data_collator,)

trainer.train()`

**Comment by linguoqi** [comment]
![6cf8b6f8b63295495b7ec7cd2af793b](https://github.com/modelscope/swift/assets/157370038/41ea7fc1-75c2-476a-b69d-ebe2d04b2894)

**Comment by tastelikefeet** [comment]
可以尝试：
```python
import os

os.environ['CUDA_VISIBLE_DEVICES'] = '3'
from modelscope import Model, AutoTokenizer
from swift import Swift, AdapterConfig, Trainer, TrainingArguments, PrefixTuningConfig
import torch
from transformers import default_data_collator

model = Model.from_pretrained('modelscope/Llama-2-7b-chat-ms', torch_dtype=torch.bfloat16)
adapter_config = AdapterConfig(
    dim=model.config.hidden_size,
    target_modules=['mlp'],
    method_name='forward',
    hidden_pos=0,
    adapter_length=32,
)

prompt_config = PrefixTuningConfig(
    num_virtual_tokens=20,  # hidden states的维度
    task_type='CAUSAL_LM'
)

print(prompt_config)
model = Swift.prepare_model(model, prompt_config)
tokenizer = AutoTokenizer.from_pretrained('modelscope/Llama-2-7b-chat-ms')

import json
from datasets import Dataset

train_ds = [json.loads(
    '{"instruction": "In the context of a black start problem, where a power grid is being restored systematically from a complete shutdown without external power sources, what is the optimal restoration order for generator bus set and bus set?", "input": "Current case has 5 generator buses, 9 load buses, and 15 lines.Generator Parameters: Generator Bus Index 1, Capacity 332.4MW, Ramp Rate 1.662MW/min. Generator Bus Index 2, Capacity 140.0MW, Ramp Rate 0.7MW/min. Generator Bus Index 3, Capacity 100.0MW, Ramp Rate 0.5MW/min. Generator Bus Index 6, Capacity 100.0MW, Ramp Rate 0.5MW/min. Generator Bus Index 8, Capacity 100.0MW, Ramp Rate 0.5MW/min.  All generator units have a grid connection time of 0 minutes.Line Network Structure: Line 0: Bus 0 -> Bus 1; Line 1: Bus 0 -> Bus 4; Line 2: Bus 1 -> Bus 2; Line 3: Bus 1 -> Bus 3; Line 4: Bus 1 -> Bus 4; Line 5: Bus 2 -> Bus 3; Line 6: Bus 3 -> Bus 4; Line 7: Bus 5 -> Bus 10; Line 8: Bus 5 -> Bus 11; Line 9: Bus 5 -> Bus 12; Line 10: Bus 8 -> Bus 9; Line 11: Bus 8 -> Bus 13; Line 12: Bus 9 -> Bus 10; Line 13: Bus 11 -> Bus 12; Line 14: Bus 12 -> Bus 13; Critical Load Buses for Restoration: [5, 13, 11]. Additional Load Buses (optional for restoration): [4, 7, 9, 10, 12, 14]. Restoration of additional buses is considered beneficial for the recovery of critical load buses and generator buses.The recovery time for each line is uniformly set at 5 minutes.", "output": "The optimal order for generator bus restoration is [1, 2, 3, 6, 8]. Furthermore, The optimal order for bus restoration is [1, 2, 3, 5, 6, 4, 7, 8, 11, 13]."}')]
train_ds = train_ds * 4
train_dataset = Dataset.from_list(train_ds)
val_dataset = Dataset.from_list(train_ds)

print(train_dataset)


def tokenize_function(examples):
    return tokenizer(examples["instruction"], examples["input"], examples["output"], padding="max_length",
                     truncation=True, max_length=128)


train_dataset = train_dataset.map(tokenize_function, batched=True)
val_dataset = val_dataset.map(tokenize_function, batched=True)

arguments = TrainingArguments(
    output_dir='./outputs',
    per_device_train_batch_size=4,
)

trainer = Trainer(model, arguments, train_dataset=train_dataset,
                  eval_dataset=val_dataset,
                  data_collator=default_data_collator, )

trainer.train()
```
